File: app/src/main/java/com/zeeshanhussain/timetable/viewmodel/LectureViewModelFactory.java
package com.zeeshanhussain.timetable.viewmodel;

import com.zeeshanhussain.timetable.database.AppDatabase;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class LectureViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private AppDatabase appDatabase;
    private int day;

    public LectureViewModelFactory(AppDatabase database, int mDay) {
        appDatabase = database;
        day = mDay;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        //noinspection unchecked
        return (T) new LectureViewModel(appDatabase, day);
    }
}

File: app/src/main/java/com/zeeshanhussain/timetable/viewmodel/LectureViewModelFactory.kt
package com.zeeshanhussain.timetable.viewmodel

import com.zeeshanhussain.timetable.database.AppDatabase

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class LectureViewModelFactory(private val appDatabase: AppDatabase, private val day: Int) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {

        return LectureViewModel(appDatabase, day) as T
    }
}
